File: AalsiTrader/Models/CommMessage.swift
import Foundation

struct CommMessage: Codable, Identifiable {
    let id: String
    let from: String
    let fromGreek: String
    let fromColor: String
    let to: String
    let toGreek: String
    let toColor: String
    let content: String
    let timestamp: ActivityTimestamp

    var date: Date {
        timestamp.asDate
    }
}

File: AalsiTraderAndroid/app/src/main/java/com/aalsitrader/android/model/CommMessage.kt
package com.aalsitrader.android.model

import kotlinx.serialization.Serializable

@Serializable
data class CommMessage(
    val id: String,
    val from: String = "",
    val fromGreek: String = "",
    val fromColor: String = "",
    val to: String = "",
    val toGreek: String = "",
    val toColor: String = "",
    val content: String = "",
    val timestamp: ActivityTimestamp = ActivityTimestamp.Epoch(0.0),
)
